src/Microsoft.AspNetCore.Mvc.Razor/Internal/ViewComponentTagHelperDescriptorFactory.cs

```diff
@@ -56,7 +57,6 @@ public IEnumerable<TagHelperDescriptor> CreateDescriptors(string assemblyName)
                     StringComparison.Ordinal));
 
             var tagHelperDescriptors = viewComponentDescriptors
-                .Where(descriptor => !descriptor.Parameters.Any(parameter => parameter.ParameterType.GetTypeInfo().IsGenericType))
```

Comment:
If we don't filter these out, when someone writes a ViewComponent like below it will show up in tag helper intellisense.
`public IViewComponentResult Invoke<T>(Dictionary<string, T> dict)`

Note: This is not a valid Invoke method even with regular ViewComponents